NuGet.config

```diff
@@ -7,6 +7,7 @@
     <add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
     <add key="dotnet-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" />
     <add key="dotnet8" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8/nuget/v3/index.json" />
+    <add key="dotnet7" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7/nuget/v3/index.json" />
```

Comment:
Why are we adding this? We shouldn't have any dependencies on .NET 7 feed.

Reply:
![image](https://github.com/dotnet/extensions/assets/4403806/0813a539-0944-4a11-b7cc-6867ed49ff5b)

@vitek-karas could you help us understand why we have .NET 7 dependency for .NET Core 3.1?

Reply:
Seems to me that this file gets generated here: https://github.com/dotnet/installer/blob/f8275a0edc0a90b593f2e3f73c4b9b3e908e3e22/src/redist/targets/GenerateBundledVersions.targets#L989-L991

Which uses this version:

https://github.com/dotnet/installer/blob/f8275a0edc0a90b593f2e3f73c4b9b3e908e3e22/eng/Versions.props#L27-L29

That is why we ended up having to add the dependency to the .net 7 feed.

Reply:
@sbomer would know

Reply:
This change now causes downstream repos to stop building unless those add a dependency on dotnet7 feed (which is a very undesirable thing):
```
D:\a\_work\1\s\src\Microsoft.Azure.Extensions.Messaging.StorageQueues\Microsoft.Azure.Extensions.Messaging.StorageQueues.csproj : error NU1603: Microsoft.Azure.Extensions.Messaging.StorageQueues depends on Microsoft.NET.ILLink.Analyzers (>= 7.0.100-1.23211.1) but Microsoft.NET.ILLink.Analyzers 7.0.100-1.23211.1 was not found. An approximate best match of Microsoft.NET.ILLink.Analyzers 8.0.100-1.22608.1 was resolved. [TargetFramework=net8.0]
##[error]src\Microsoft.Azure.Extensions.Messaging.StorageQueues\Microsoft.Azure.Extensions.Messaging.StorageQueues.csproj(0,0): error NU1603: (NETCORE_ENGINEERING_TELEMETRY=Build) Microsoft.Azure.Extensions.Messaging.StorageQueues depends on Microsoft.NET.ILLink.Analyzers (>= 7.0.100-1.23211.1) but Microsoft.NET.ILLink.Analyzers 7.0.100-1.23211.1 was not found. An approximate best match of Microsoft.NET.ILLink.Analyzers 8.0.100-1.22608.1 was resolved.
```

Reply:
Thank you @sbomer. Should the packages be published to dotnet8 (or perhaps dotnet-public) feed, or is there an expectation that consumers will be required to pull from dotnet7 feed?

Reply:
I may of course be wrong here so I'll wait for @marcpopMSFT to answer the above question, but IMO any packages that are produced by .NET and may be required by the .NET 8 SDK and is not already in the dotnet-public feed (meaning it's not yet in NuGet), then I believe it should be in the dotnet8 feed. It doesn't _"feel right"_ to have to depend on the dotnet7 feed when using the latest .NET 8 SDK.